pydoctor/templatewriter/writer.py

```diff
@@ -82,22 +82,34 @@ def writeIndividualFiles(self, obs: Iterable[model.Documentable]) -> None:
 
     def writeSummaryPages(self, system: model.System) -> None:
         import time
-        for pclass in summary.summarypages:
+        for pclass in summary.summaryPages(system):
             system.msg('html', 'starting ' + pclass.__name__ + ' ...', nonl=True)
             T = time.time()
             page = pclass(system=system, template_lookup=self.template_lookup)
             with self.build_directory.joinpath(pclass.filename).open('wb') as fobj:
                 flattenToFile(fobj, page)
             system.msg('html', "took %fs"%(time.time() - T), wantsnl=False)
 
+        if len(system.root_names) == 1:
+            # If there is just a single root module it is written to index.html to produce nicer URLs.
+            # To not break old links we also create a symlink from the full module name to the index.html
+            # file. This is also good for consistency: every module is accessible by <full module name>.html
+            root_module_path = (self.build_directory / (list(system.root_names)[0] + '.html'))
+            try:
+                root_module_path.unlink()
+                # not using missing_ok=True because that was only added in Python 3.8 and we still support Python 3.6
+            except FileNotFoundError:
+                pass
+            root_module_path.symlink_to('index.html')
+
     def _writeDocsFor(self, ob: model.Documentable) -> None:
         if not ob.isVisible:
             return
         if ob.documentation_location is model.DocLocation.OWN_PAGE:
             if self.dry_run:
                 self.total_pages += 1
             else:
-                with self.build_directory.joinpath(f'{ob.fullName()}.html').open('wb') as fobj:
+                with self.build_directory.joinpath(ob.url.split('#')[0]).open('wb') as fobj:
```

Comment:
Not sure we absolutely need to split on `#`. Since `ob.documentation_location is model.DocLocation.OWN_PAGE`, `page_object` will return `self` and nothing will be added after the filename in the URL. 

I think the following is OK:

```suggestion
                # Documentable.page_object.url will alway be a plain filename.
                with self.build_directory.joinpath(ob.page_object.url).open('wb') as fobj:
```

Reply:
I think splitting here is cleaner, because otherwise there is an undocumented constraint on the `url` method. If you really want to avoid that split we'd need to document that in the `url` method that it should not set an anchor for "page objects".

Reply:
Yes, to mt it's pretty clear the that the url method that it should not set an anchor for "page objects".

Reply:
Right. I still prefer such assumptions to be documented. I feel like not depending on such behavior is even better.

Reply:
You could quickly add a note to the url property docstring saying that page object should always return a plain relative file name, without anchor.

Reply:
Done.